### PR TITLE
CLDR-17143 Deprecate collation type reformed, remove it from coverage and locale data

### PR DESCRIPTION
**Pull request by pedberg-icu**
CLDR-17143

- [x] This PR completes the ticket.

Mark collation type reformed as deprecated, remove it from coverage and locale data

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Comment by pedberg-icu** [comment]
Working on unit test failures (some were masked by unrelated NPEs)

File: common/main/en.xml [hunk 1]
```diff
@@ -1376,7 +1376,6 @@ annotations.
 			<type key="collation" type="phonebook">Phonebook Sort Order</type>
 			<type key="collation" type="phonetic">Phonetic Sort Order</type>
 			<type key="collation" type="pinyin">Pinyin Sort Order</type>
-			<type key="collation" type="reformed">Reformed Sort Order</type>
```
Comment (markusicu) [inline common/main/en.xml:1379]: On one hand, it seems like we might want to keep display names until some time (maybe 1 year?) after their item starts being deprecated.
On the other hand, I suspect that this collation type and its display names are rarely used... so fine with me.
Is it normal for CLDR to remove display data immediately when deprecating something?

**Review by markusicu — COMMENTED** [review]
changes lgtm
looking forward to test fixes

File: common/main/en.xml (hunk 1 above)
Comment (pedberg-icu) [reply common/main/en.xml:1379]: CLDR stopped using the "reformed" type in CLDR 42. We decided in TC today that it was okay to remove the translations in this PR.

**Review by markusicu — APPROVED** [review]

**Pull request merged**